sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -89,7 +89,7 @@ object ResolveHints {
       case h: UnresolvedHint if BROADCAST_HINT_NAMES.contains(h.name.toUpperCase(Locale.ROOT)) =>
         if (h.parameters.isEmpty) {
           // If there is no table alias specified, turn the entire subtree into a BroadcastHint.
-          ResolvedHint(h.child, HintInfo(isBroadcastable = Option(true)))
+          ResolvedHint(h.child, HintInfo(forceBroadcast = true))
```

Comment:
We are not forcing the broadcast. In the join strategy, we still check the other conditions to determine whether one table can be broadcasted.